src/vivarium/framework/event.py

```diff
@@ -111,7 +90,7 @@ def __init__(self, manager):
         self.manager = manager
         self.listeners = [[] for _ in range(10)]
 
-    def emit(self, event: Event) -> Event:
+    def emit(self, index: pd.Index, user_data: Dict = {}) -> Event:
```

Comment:
I think you still want this default to be None because you don't want something mutable here but then in the function itself, if user_data comes in as None, reassign it to an empty dict